dbms/src/Storages/BackgroundProcessingPool.cpp

```diff
@@ -87,6 +87,9 @@ BackgroundProcessingPool::BackgroundProcessingPool(int size_)
     : size(size_)
     , thread_ids_counter(size_)
 {
+    if (size <= 0)
+        throw Exception("BackgroundProcessingPool size must be greater than 0", ErrorCodes::LOGICAL_ERROR);
```

Comment:
maybe set `size` to be 1 when `size <= 0` with a warnings logging is more robust? For example, run mock test without starting the tiflash-proxy

Reply:
because `size` is a const value that can not be assign 😄, but IMO it is not a good design and we should implement a function to support change the size of thread pool. cc @JinheLin

Reply:
but I set the initial value of server_info.cpu_info.logical_cores to `std::thread::hardware_concurrency`, so even without tiflash-proxy, the input value would not be 0.

Reply:
resolved as the default value of `logical_cores` is `std::thread::hardware_concurrency()`